src/index.js

```diff
@@ -59,6 +83,114 @@ document.addEventListener('DOMContentLoaded', () => {
+  }else{
+    return undefined
+  }
+}
+
+const allParadigmSizes = ['BASIC', 'FULL', 'LINGUISTIC']
+
+
+/**
+ * cycles between BASIC, FULL, LINGUISTIC
+ *
+ * @param {String} size
+ * @return {String}
+ */
+function getNextParadigmSize(size) {
+  return allParadigmSizes[(allParadigmSizes.indexOf(size) + 1) % allParadigmSizes.length]
+}
+
+/**
+ * https://stackoverflow.com/a/11654596
+ * get the current url and update a query param
+ *
+ * @param {String} key
+ * @param {String} value
+ * @returns {String}
+ */
+function updateQueryParam(key, value) {
+  let url = window.location.href
+  let re = new RegExp('([?&])' + key + '=.*?(&|#|$)(.*)', 'gi'),
+    hash
+
+  if (re.test(url)) {
+    if (typeof value !== 'undefined' && value !== null) {
+      return url.replace(re, '$1' + key + '=' + value + '$2$3')
+    } else {
+      hash = url.split('#')
+      url = hash[0].replace(re, '$1$3').replace(/([&?])$/, '')
+      if (typeof hash[1] !== 'undefined' && hash[1] !== null) {
+        url += '#' + hash[1]
+      }
+      return url
+    }
+  } else {
+    if (typeof value !== 'undefined' && value !== null) {
+      const separator = url.indexOf('?') !== -1 ? '&' : '?'
+      hash = url.split('#')
+      url = hash[0] + separator + key + '=' + value
+      if (typeof hash[1] !== 'undefined' && hash[1] !== null) {
+        url += '#' + hash[1]
+      }
+      return url
+    } else {
+      return url
+    }
+  }
+}
+
+
+/**
+ * attach handlers to the "show more/less" button. So that it:
+ *
+ *  - loads a more detailed paradigm when clicked or do nothing and report to console if the request for the paradigm errors
+ *  - changes its text to "show less" when the paradigm has the largest size
+ *    and shows the smallest paradigm when clicked
+ *  - prepare the new "show more/less" button to do these 3
+ */
+function setupParadigmSizeToggleButton() {
+  const toggleButton = document.getElementsByClassName('paradigm__size-toggle-button')[0]
+
+  const nextParadigmSize = getNextParadigmSize(paradigmSize)
+  toggleButton.addEventListener('click', () => {
+    fetch(Urls['cree-dictionary-lemma-detail']() + `?lemma-id=${lemmaId}&paradigm-size=${nextParadigmSize}`).then(r => {
+      if (r.ok) {
+        return r.text()
+      } else {
+        throw new Error(`${r.status} ${r.statusText} when loading paradigm: ${r.text()}`)
+      }
+    }).then(
+      text => {
+        const paradigmFrag = document.createRange().createContextualFragment(text)
+        if (allParadigmSizes.indexOf(nextParadigmSize) === allParadigmSizes.length - 1) {
+          paradigmFrag.querySelector('.paradigm__size-toggle-button-text').textContent = 'show less'
+          paradigmFrag.querySelector('.paradigm__size-toggle-plus-minus').textContent = '- '
+
+        } else {
+          paradigmFrag.querySelector('.paradigm__size-toggle-button-text').textContent = 'show more'
+          paradigmFrag.querySelector('.paradigm__size-toggle-plus-minus').textContent = '+ '
+        }
```

Comment:
I wish this was all handled on the server side or something, but I don't really have a good solution for that right now :/

Reply:
if you visit with e.g. /word/xxxx?paradigm-size=LINGUSTIC this is indeed rendered on the server side. so currently it's a mixed solution